src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -114,29 +133,62 @@ module.exports = function<T, P, I>(config : HostConfig<T, P, I>) : Reconciler {
       const next = completeWork(current, workInProgress);
+
+      // The work is now done. We don't need this anymore. This flags
+      // to the system not to redo any work here.
+      workInProgress.pendingProps = null;
+
+      // TODO: Stop using the parent for this purpose. I think this will break
+      // down in edge cases because when nodes are reused during bailouts, we
+      // don't know which of two parents was used. Instead we should maintain
+      // a temporary manual stack.
+      // $FlowFixMe: This downcast is not safe. It is intentionally an error.
+      const parent = workInProgress.parent;
+
+      // Ensure that remaining work priority bubbles up.
+      if (parent && workInProgress.pendingWorkPriority !== NoWork &&
+          (parent.pendingWorkPriority === NoWork ||
+          parent.pendingWorkPriority > workInProgress.pendingWorkPriority)) {
+        parent.pendingWorkPriority = workInProgress.pendingWorkPriority;
+      }
+
       if (next) {
         // If completing this work spawned new work, do that next.
         return next;
       } else if (workInProgress.sibling) {
         // If there is more work to do in this parent, do that next.
         return workInProgress.sibling;
-      } else if (workInProgress.parent) {
+      } else if (parent) {
         // If there's no more work in this parent. Complete the parent.
-        // TODO: Stop using the parent for this purpose. I think this will break
-        // down in edge cases because when nodes are reused during bailouts, we
-        // don't know which of two parents was used. Instead we should maintain
-        // a temporary manual stack.
-        // $FlowFixMe: This downcast is not safe. It is intentionally an error.
-        workInProgress = workInProgress.parent;
+        workInProgress = parent;
       } else {
         // If we're at the root, there's no more work to do.
-        currentRootsWithPendingWork = null;
-        return null;
+        console.log('completed flush with remaining work at priority', workInProgress.pendingWorkPriority);
+        if (workInProgress.pendingWorkPriority !== NoWork) {
+          // TODO: This removes all but one node. Broken.
+          currentRootsWithPendingWork = workInProgress;
+          const nextWork = findNextUnitOfWork();
+          if (!nextWork) {
+            // Something went wrong and there wasn't actually any more work.
+            // TODO: This currently fails because of the parent/root hacks.
+            // throw new Error('Should never finish with a priority level unless there is work.');
+            currentRootsWithPendingWork = null;
+            return null;
+          }
+          return nextWork;
+        } else {
+          currentRootsWithPendingWork = null;
+          return null;
+        }
       }
     }
   }
 
   function performUnitOfWork(workInProgress : Fiber) : ?Fiber {
+    // Ignore work if there is nothing to do.
+    if (workInProgress.pendingProps === null) {
```

Comment:
There is a known problem if you traverse back out of the update of a root, then the next siblings won't have any work to do so they just get ignored here. However, they may also have work for the wrong priority!

I'm also not sure if you could potentially get here if there is a deeper update below. Presumably that only happens in the bail out case which is already covered.

But yea, this should be an error if the other things are fixed.